front-end/src/routes.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import { useEffect } from "react";
+import { useLocation } from "react-router";
+import { RouteObject, useNavigate } from "react-router-dom";
+import { Article } from "./administration/Article";
+import { User } from "./administration/User";
+import { Analytics } from "./core/Analytics";
+import { Request as PassphraseResetRequest } from "./identification/Passphrase/Reset/Request";
+import { Confirm as PassphraseResetConfirm } from "./identification/Passphrase/Reset/Confirm";
+import { Change as PassphraseChange } from "./identification/Passphrase/Change";
+import { Register } from "./identification/Register";
+import { Arad } from "./Arad";
+import { Login } from "./identification/Login";
+import { Redirect } from "./core/Redirect";
+
+export enum RoutesEnum {
+  HOME,
+  REGISTER,
+  LOGIN,
+  PASSPHRASE,
+  RESET,
+  CONFIRM,
+  SEARCH,
+  ARTICLES,
+  USERS,
+  CODE,
+}
+
+const ROUTE_MAP: {
+  [key in RoutesEnum]: RouteObject & {
+    element: JSX.Element;
+    isProtected?: boolean;
+  };
+} = {
+  [RoutesEnum.HOME]: {
+    path: "/",
+    element: <Arad />,
+  },
+  [RoutesEnum.REGISTER]: {
+    path: "register",
+    element: <Register />,
+  },
+  [RoutesEnum.LOGIN]: {
+    path: "login",
+    element: <Login />,
+  },
+  [RoutesEnum.PASSPHRASE]: {
+    path: "passphrase",
+    element: <PassphraseChange />,
+  },
+  [RoutesEnum.RESET]: {
+    path: "reset",
+    element: <PassphraseResetRequest />,
+  },
+  [RoutesEnum.CONFIRM]: {
+    path: "confirm",
+    element: <PassphraseResetConfirm />,
+  },
+  [RoutesEnum.SEARCH]: {
+    path: "search/:articleId",
+    element: <Analytics />,
+    isProtected: true,
+  },
+  [RoutesEnum.ARTICLES]: {
+    path: "articles/:articleId",
+    element: <Article />,
+    isProtected: true,
+  },
+  [RoutesEnum.USERS]: {
+    path: "users/:userId",
+    element: <User />,
+    isProtected: true,
```

Comment:
What about like, /search (list articles from search view) /articles (list articles from admin view for bulk action) /users (list users)? Are those omitted here or does this logic somehow allow rendering those routes?

Reply:
Yeah -- omitted. I hadn't used Outlet before and wasn't entirely sure what was going on. I'll refactor this tmmrw to account for those things